corehq/messaging/scheduling/models/timed_schedule.py

```diff
@@ -0,0 +1,197 @@
+
+        instance.next_event_due = (
+            UserTime(user_timestamp, instance.timezone)
+            .server_time()
+            .done()
+            .replace(tzinfo=None)
+        )
+
+    def get_current_event_content(self, instance):
+        current_event = self.memoized_events[instance.current_event_num]
+        return current_event.memoized_content
+
+    def move_to_next_event(self, instance):
+        instance.current_event_num += 1
+        if instance.current_event_num >= len(self.memoized_events):
+            instance.schedule_iteration_num += 1
+            instance.current_event_num = 0
+        self.set_next_event_due_timestamp(instance)
+
+        if (
+            self.total_iterations != self.REPEAT_INDEFINITELY and
+            instance.schedule_iteration_num > self.total_iterations
+        ):
+            instance.active = False
+
+    @classmethod
+    def create_simple_daily_schedule(cls, domain, time, content, total_iterations=REPEAT_INDEFINITELY):
+        schedule = cls(domain=domain)
+        schedule.set_simple_daily_schedule(time, content, total_iterations=total_iterations)
+        return schedule
+
+    def set_simple_daily_schedule(self, time, content, total_iterations=REPEAT_INDEFINITELY):
+        with transaction.atomic():
+            self.schedule_length = 1
+            self.total_iterations = total_iterations
+            self.save()
+
+            for event in self.timedevent_set.all():
+                event.content.delete()
+                event.delete()
+
+            if content.pk is None:
+                content.save()
+
+            event = TimedEvent(
+                schedule=self,
+                order=1,
+                day=0,
+                time=time
+            )
+            event.content = content
+            event.save()
+
+    @classmethod
+    def create_simple_monthly_schedule(cls, domain, time, days, content, total_iterations=REPEAT_INDEFINITELY):
+        schedule = cls(domain=domain)
+        schedule.set_simple_monthly_schedule(time, days, content, total_iterations=total_iterations)
+        return schedule
+
+    def set_simple_monthly_schedule(self, time, days, content, total_iterations=REPEAT_INDEFINITELY):
+        with transaction.atomic():
+            self.schedule_length = self.MONTHLY
+            self.total_iterations = total_iterations
+            self.save()
+
+            for event in self.timedevent_set.all():
+                event.content.delete()
+                event.delete()
+
+            if content.pk is None:
+                content.save()
+
+            order = 1
+            for day in days:
+                event = TimedEvent(
+                    schedule=self,
+                    order=order,
+                    day=day,
+                    time=time
+                )
+
+                if order > 1:
+                    content.pk = None
+                    content.save()
```

Comment:
So this duplicates the content for each event. Would it be bad to have them all pointing at the same content object?